qiskit/aqua/operators/state_functions/circuit_state_fn.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" An Object to represent State Functions constructed from Operators """
+
+from typing import Union
+import numpy as np
+
+from qiskit import QuantumCircuit, BasicAer, execute
+from qiskit.circuit import Instruction, ParameterExpression
+from qiskit.extensions import Initialize, IGate
+
+from ..operator_base import OperatorBase
+from ..combo_operators import SummedOp
+from .state_fn import StateFn
+
+
+class CircuitStateFn(StateFn):
+    """ A class for representing state functions and measurements.
```

Comment:
This lengthy description seems to be copied in a bunch of these. Would it not be better just to have this in the base class once and then state anything specific to the subclass in each of the subclasses. Actually having gone further I see it is in the base class along with docstrings for the methods defined there like to_density_matrix and to_matrix which I commented on below. The docstring from the base class I believe should be fine and does not need to be replicated everywhere, which is a maintenance nightmare, assuming it equally applies to the sub-classes which I guess it should

Reply:
Fixed